### PR TITLE
add shortcuts to '{select/expand}ToMatching'

### PR DESCRIPTION
**Pull request by kevinushey**
Note: `Ctrl+Shift+P` and `Ctrl+Alt+Shift+P` chosen since `Ctrl+P` is already bound to `jumpToMatching()` and these are effectively suped-up versions of that behaviour.

### CODE REVIEW

File: src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml [hunk 1]
```diff
@@ -427,6 +427,8 @@ well as menu structures (for main menu and popup menus).
          <shortcut refid="jumpTo" value="Shift+Alt+J" if="!org.rstudio.core.client.BrowseCap.isMacintosh()"/>
          <shortcut refid="jumpTo" value="Cmd+Shift+Alt+J" if="org.rstudio.core.client.BrowseCap.isMacintosh()"/>
          <shortcut refid="jumpToMatching" value="Ctrl+P"/>
+         <shortcut refid="selectToMatching" value="Ctrl+Shift+P"/>
+         <shortcut refid="expandToMatching" value="Ctrl+Shift+Alt+P"/>
```
Comment (jjallaire) [inline src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml]: On the Mac Ctrl+Shift+Alt+P inserts a special character for me so I don't think we can use it.

Comment (kevinushey) [inline src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml]: Really? Darn, I don't get any special character with that key combination on my Mac (maybe a difference in settings somewhere?)

We can think about a different key combination later.

**Comment by kevinushey** [comment]
@jmcphers, I think this is ready to merge, but I've overridden your change re: using `execCommand` and instead explicitly ensure the cursor is visible after a `{select/jump/expand}ToMatching`.

I'm guessing you know, but the `execCommand` works is that the default 'scroll' behaviour for those commands is to scroll (animate) to the cursor position; we simply take control of that ourselves and ensure the cursor is visible (although without animation). Do you think this is okay?

**Comment by jmcphers** [comment]
LGTM

**Pull request merged**